core/json/src/main/scala/net/liftweb/json/JsonAST.scala

```diff
@@ -19,32 +19,97 @@ package json
 
 import scala.language.implicitConversions
 
+/**
+ * This object contains the abstract snytax tree (or AST) for working with JSON objects in lift-json.
+ *
+ * The purpose of the JsonAST is to represent and manipulate JSON by leveraging Scala language features like types,
+ * case classes, etc. The AST should allow you to represent anything you could imagine from JSON land using the Scala
+ * type system.
+ *
+ * Everything in the AST has a single root: JValue. A JValue could, quite literally, be anything. It could be an
+ * an object (represented by [[JObject]]), a string ([[JString]]), a null ([[JNull]]), and so on. So, when constructing
```

Comment:
In #1650 I both linked and monospaced all class names (via backticks). Pretty easy change, just sticking backticks outside of the braces.